### DISCUSSION TITLE
Question regarding BindingState wrapper

### DISCUSSION THREAD

**Discussion by jkay-kim** [pointfreeco/swift-composable-architecture#2306] [Q&A] [answered]
```swift
struct SomeFeature: ReducerProtocol {
    struct State: Equatable {
        @BindingState var someText: String = ""
    }
    
    enum Action: BindableAction {
        case binding(BindingAction<State>)
    }
    
    func reduce(into state: inout State, action: Action) -> EffectTask<Action> {
        switch action {
        case .binding(\.someText):
            return .none
            
        case .binding:
            return .none
        }
    }
}

```

Hello. I am new to TCA.
When I try to bind a TextField from a view to `viewStore.$someText`, the state of 'someText' is not updated. It remains the initial value "".
I think there is no logic to set the new text value to the `someText` variable, such as state.someText = newValue. 
How can I solve the problem?

Thank you 🙇‍♂️

**Comment by mbrandonw** [comment, marked as answer]
Hi @jkay-kim, yes something is missing from your code. You should have seen a purple warning in Xcode letting you know that something was wrong when you typed into the text field. Did you not?

To fix this you have to use the [`BindingReducer`](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/bindingreducer). We have an [article](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/bindings) in the documentation that describes this, but in your case you want this:

```swift
struct SomeFeature: ReducerProtocol {
  struct State: Equatable {
    @BindingState var someText: String = ""
  }
  
  enum Action: BindableAction {
    case binding(BindingAction<State>)
  }
  
  var body: some ReducerProtocolOf<Self> {
    BindingReducer()
    
    Reduce { state, action in 
      switch action {
      case .binding(\.someText):
        return .none
          
      case .binding:
        return .none
      }
    }
  }
}
```

**Comment by jkay-kim** [comment]
@mbrandonw Thank you for your answer.
I was misunderstanding the concept of Reducer body & Reduce function that Reduce function is required func.
After moving implementation of reduce function to reducer body including BindingReducer() as you suggested, it works like a magic. Thank you!
